### PR TITLE
[ci] Publish serving tar and zip

### PR DESCRIPTION
**Pull request by xyang16**
## Description ##

Brief description of what this PR is about

Publish serving tar and zip

### CODE REVIEW

File: .github/workflows/serving_publish.yml [hunk 1]
```diff
@@ -58,17 +58,23 @@ jobs:
           aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
           aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
           aws-region: us-east-2
-      - name: Publish snapshot deb
+      - name: Publish snapshot deb, tar and zip
         if: ${{ github.event.inputs.mode == '' || github.event.inputs.mode == 'snapshot' }}
         run: |
+          ./gradlew :serving:distZip -Psnapshot
```
Comment (frankfliu) [inline .github/workflows/serving_publish.yml:64]: We don't need publish snapshot .zip and .tar file

Comment (xyang16) [reply .github/workflows/serving_publish.yml:64]: Fixed.

File: .github/workflows/serving_publish.yml [hunk 2]
```diff
@@ -58,17 +58,23 @@ jobs:
...
           aws s3 cp serving/build/distributions/*.deb s3://djl-ai/publish/djl-serving/
+          aws s3 cp serving/build/distributions/*.tar s3://djl-ai/publish/djl-serving/
+          aws s3 cp serving/build/distributions/*.zip s3://djl-ai/publish/djl-serving/
           aws cloudfront create-invalidation --distribution-id E371VB8JQ6NRVY --paths "/djl-serving/djl-serving_${DJL_VERSION}*"
-      - name: Publish release deb
+      - name: Publish release deb, tar and zip
         if: ${{ github.event.inputs.mode == 'staging' }}
         run: |
+          ./gradlew :serving:distZip -Pstaging
           ./gradlew :serving:createDeb -Pstaging
```
[the remark this answers is not in the document]
Comment (xyang16) [reply .github/workflows/serving_publish.yml:72]: Fixed

File: .github/workflows/serving_publish.yml [hunk 3]
```diff
@@ -61,14 +61,17 @@ jobs:
...
         if: ${{ github.event.inputs.mode == 'staging' }}
         run: |
-          ./gradlew :serving:createDeb -Pstaging
+          ./gradlew :serving:dZ :serving:createDeb -Pstaging
           DJL_VERSION=$(cat gradle.properties | awk -F '=' '/djl_version/ {print $2}')
           aws s3 cp serving/build/distributions/*.deb s3://djl-ai/publish/djl-serving/
           aws cloudfront create-invalidation --distribution-id E371VB8JQ6NRVY --paths "/djl-serving/djl-serving_${DJL_VERSION}*"
+          aws s3 cp serving/build/distributions/*.tar s3://djl-ai/publish/djl-serving/
+          aws s3 cp serving/build/distributions/*.zip s3://djl-ai/publish/djl-serving/
+          aws cloudfront create-invalidation --distribution-id E371VB8JQ6NRVY --paths "/djl-serving/serving-${DJL_VERSION}*"
```
Comment (frankfliu) [inline .github/workflows/serving_publish.yml:77]: We only need invalidate once

File: .github/workflows/serving_publish.yml [hunk 4]
```diff
@@ -61,14 +61,17 @@ jobs:
       - name: Publish snapshot deb
         if: ${{ github.event.inputs.mode == '' || github.event.inputs.mode == 'snapshot' }}
         run: |
-          ./gradlew :serving:createDeb -Psnapshot
+          ./gradlew :serving:distZip -Psnapshot
           DJL_VERSION=$(cat gradle.properties | awk -F '=' '/djl_version/ {print $2}')
           aws s3 cp serving/build/distributions/*.deb s3://djl-ai/publish/djl-serving/
-          aws cloudfront create-invalidation --distribution-id E371VB8JQ6NRVY --paths "/djl-serving/djl-serving_${DJL_VERSION}*"
-      - name: Publish release deb
+          aws cloudfront create-invalidation --distribution-id E371VB8JQ6NRVY --paths "/djl-serving/djl-serving_${DJL_VERSION}~SNAPSHOT*"
```
Comment (frankfliu) [inline .github/workflows/serving_publish.yml:67]: We don't need add ~SNAPSHOT here, none SNAPSHOT version doesn't exist yet

Comment (xyang16) [reply .github/workflows/serving_publish.yml:67]: I can see non SNAPSHOT version like: [djl-serving_0.19.0-1_all.deb](https://s3.console.aws.amazon.com/s3/object/djl-ai?region=us-east-1&prefix=publish/djl-serving/djl-serving_0.19.0-1_all.deb)

Comment (frankfliu) [reply .github/workflows/serving_publish.yml:67]: once we published 0.19.0, the snapshot build will be 0.20.0

Comment (xyang16) [reply .github/workflows/serving_publish.yml:67]: ok

File: .github/workflows/serving_publish.yml (hunk 3 above)
Comment (xyang16) [reply .github/workflows/serving_publish.yml:77]: One is djl-serving_, one is serving-

**Review by frankfliu — APPROVED** [review]

**Pull request merged**